### PR TITLE
feat(routeTable): ignoreRoutes to use route resource 

### LINKED ISSUE #1036: RouteTable to VPCEndpoint reference
### What problem are you facing?

No ability to attach a VPCEndpoint to RouteTables routes. 

### How could Crossplane help solve your problem?

Make new VPCEndpoints a referenced gateway type of thing in the routes of RouteTable.

### PR DESCRIPTION
**Pull request by haarchri**
Signed-off-by: haarchri <chhaar30@googlemail.com>

### Description of your changes
- added in `RouteTable.ec2.aws.crossplane.io/v1beta1`  a new field `ignoreRoutes: true `  to toggle the reconciliation of the routes array. Existing users wouldn't be affected and new users of RouteTable would set this to false and use inline routes / routes objects.
- added RouteTableRef/Selector in `Route.ec2.aws.crossplane.io/v1alpha1`
- changed all referencers/resolvers in ec2 v1beta1 to generated resolvers
- added new examples

Fixes #1036 #657 #1037

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested
- create all resources and check status & up2date loops

```
NAME                                            READY   SYNCED   ID                      CIDR             IPV6CIDR   AGE
vpc.ec2.aws.crossplane.io/sample-vpc            True    True     vpc-0a809f17c9092bc64   10.0.0.0/16                 160m
vpc.ec2.aws.crossplane.io/sample-vpc2           True    True     vpc-0314043ee67ff9321   10.1.0.0/16                 160m

NAME                                               READY   SYNCED   ID                         VPC                     CIDR               AGE
subnet.ec2.aws.crossplane.io/sample-subnet1        True    True     subnet-0bf2f58645a1d813f   vpc-0a809f17c9092bc64   10.0.1.0/24        160m
subnet.ec2.aws.crossplane.io/sample-subnet2        True    True     subnet-03b419f501739c924   vpc-0a809f17c9092bc64   10.0.2.0/24        160m

NAME                                       READY   SYNCED   EXTERNAL-NAME
transitgateway.ec2.aws.crossplane.io/tgw   True    True    tgw-09525e8a7faab2f39

NAME                                                               READY   SYNCED   EXTERNAL-NAME
transitgatewayvpcattachment.ec2.aws.crossplane.io/tgw-vpc-attach   True    True     tgw-attach-0a26d607ca1bc2beb

NAME                                                               READY   SYNCED   ID                      VPC                     AGE
routetable.ec2.aws.crossplane.io/sample-routetable-ignore-routes   True    True    rtb-06dee28feef2c86ea   vpc-0a809f17c9092bc64   6m11s
routetable.ec2.aws.crossplane.io/sample-routetable                 True    True    rtb-0cb0739d9ed82cd1b   vpc-0a809f17c9092bc64   6m11s

NAME                                  READY   SYNCED   EXTERNAL-NAME
route.ec2.aws.crossplane.io/example   True    True     example

NAME                                                       READY   SYNCED   EXTERNAL-NAME
vpcendpoint.ec2.aws.crossplane.io/sample-vpcendpoint-rtb   True    True     vpce-0ecc812ddc03958a8
```

```
kubectl describe routetables sample-routetable-ignore-routes
Name:         sample-routetable-ignore-routes
Namespace:    
Labels:       <none>
Annotations:  crossplane.io/external-create-pending: 2022-01-23T14:46:48+01:00
              crossplane.io/external-create-succeeded: 2022-01-23T14:46:49+01:00
              crossplane.io/external-name: rtb-06dee28feef2c86ea
API Version:  ec2.aws.crossplane.io/v1beta1
Kind:         RouteTable
Metadata:
  Creation Timestamp:  2022-01-23T13:46:48Z
  Finalizers:
    finalizer.managedresource.crossplane.io
  Generation:  4
  Resource Version:  1402559
  UID:               f89d75d4-9e09-41a5-927a-7d155e26ed71
Spec:
  Deletion Policy:  Delete
  For Provider:
    Associations:
      Subnet Id:  subnet-03b419f501739c924
      Subnet Id Ref:
        Name:       sample-subnet2
    Ignore Routes:  true
    Region:         us-east-1
    Vpc Id:         vpc-0a809f17c9092bc64
    Vpc Id Ref:
      Name:  sample-vpc
  Provider Config Ref:
    Name:  default
Status:
  At Provider:
    Associations:
      Association Id:  rtbassoc-0631a558869eee89e
      Main:            false
      State:           associated
      Subnet Id:       subnet-03b419f501739c924
    Owner Id:          255932642927
    Route Table Id:    rtb-06dee28feef2c86ea
    Routes:
      Destination Cidr Block:  10.0.0.0/16
      Gateway Id:              local
      State:                   active
      Destination Cidr Block:  172.16.0.0/12
      State:                   active
      Transit Gateway Id:      tgw-09525e8a7faab2f39
      Gateway Id:              vpce-0ecc812ddc03958a8
      State:                   active
  Conditions:
    Last Transition Time:  2022-01-23T13:46:49Z
    Reason:                Available
    Status:                True
    Type:                  Ready
    Last Transition Time:  2022-01-23T13:55:37Z
    Reason:                ReconcileSuccess
    Status:                True
    Type:                  Synced
Events:
  Type     Reason                         Age                  From                                      Message
  ----     ------                         ----                 ----                                      -------
  Normal   CreatedExternalResource        9m22s                managed/routetable.ec2.aws.crossplane.io  Successfully requested creation of external resource
  Normal   UpdatedExternalResource        9m21s                managed/routetable.ec2.aws.crossplane.io  Successfully requested update of external resource
```

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: apis/ec2/v1beta1/routetable_types.go [hunk 1]
```diff
@@ -171,11 +175,16 @@ type RouteTableParameters struct {
...
+	IgnoreRoutes *bool `json:"ignoreRoutes,omitempty"`
+
 	// The associations between the route table and one or more subnets.
 	Associations []Association `json:"associations"`
 
-	// the routes in the route table
-	Routes []RouteBeta `json:"routes"`
+	// inline routes in the route table
+	// +optional
+	Routes []RouteBeta `json:"routes,omitempty"`
```
Comment (muvaf) [inline apis/ec2/v1beta1/routetable_types.go:187]: Can you mark this field and the type as deprecated?

File: pkg/clients/ec2/routetable.go [hunk 2]
```diff
@@ -198,6 +206,7 @@ func IsRtUpToDate(p v1beta1.RouteTableParameters, rt ec2types.RouteTable) (bool,
 	return cmp.Equal(&v1beta1.RouteTableParameters{}, patch,
```
Comment (muvaf) [inline pkg/clients/ec2/routetable.go:206]: Does that make sure to not check the routes?

File: apis/ec2/v1beta1/routetable_types.go (hunk 1 above)
Comment (haarchri) [reply apis/ec2/v1beta1/routetable_types.go:187]: what did we need to do for deprecation ?

**Comment by Dkaykay** [comment]
Hi all, 
any Updates on the status of this PR?

File: apis/ec2/v1beta1/routetable_types.go (hunk 1 above)
Comment (haarchri) [reply apis/ec2/v1beta1/routetable_types.go:187]: @muvaf  did we have an example for deprecation ?

File: pkg/clients/ec2/routetable.go (hunk 2 above)
Comment (haarchri) [reply pkg/clients/ec2/routetable.go:206]: added checks in CreateRTPatch to skip routes - does this fits ? or did you want this on a other code block ?

File: apis/ec2/v1beta1/routetable_types.go (hunk 1 above)
Comment (Dkaykay) [reply apis/ec2/v1beta1/routetable_types.go:187]: @haarchri @muvaf Is this the way https://github.com/crossplane/provider-aws/blob/master/apis/cache/v1beta1/replication_group_types.go#L355 ?

File: pkg/clients/ec2/routetable.go (hunk 2 above)
Comment (haarchri) [reply pkg/clients/ec2/routetable.go:206]: @muvaf did you want to have the check earlier ?

File: apis/ec2/v1beta1/routetable_types.go (hunk 1 above)
Comment (muvaf) [reply apis/ec2/v1beta1/routetable_types.go:187]: Standard Golang deprecation notice: https://github.com/golang/go/wiki/Deprecated

**Review by muvaf — APPROVED** [review]
LGTM!

**Comment by haarchri** [comment]
@Dkaykay  sorry for delay - i added the deprecation notice

**Pull request merged**